components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -3324,4 +3266,15 @@ private String resolveOrganizationId(String tenantDomain) throws IdentityOAuth2E
                     tenantDomain, e);
         }
     }
+
+    private String getRootTenantDomainByOrganizationId(String organizationId) {
+
+        try {
+            String rootOrgID = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
+                    .getPrimaryOrganizationId(organizationId);
+            return OAuthComponentServiceHolder.getInstance().getOrganizationManager().resolveTenantDomain(rootOrgID);
+        } catch (OrganizationManagementException e) {
+            throw new RuntimeException(e);
```

Comment:
Shall we return an IdentityOAuth2Exception

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2504/commits/13de5fc1f8327d2e23a21a5655193cb857635a69